From 861d49c2a11c1a7f0fa49d0fcdbc8dfd22342e0a Mon Sep 17 00:00:00 2001
From: Jasper St. Pierre <jstpierre@mecheye.net>
Date: Tue, 18 Dec 2012 13:02:29 +0000
Subject: screen: Increase typical alt-tab icon size to 96

Now we removed thumbnailing in the alt-tab switcher, the switcher is
more difficult to use with its small icons.

We switch to mutter-like higher resolution icons to be able to distinct
them better (the new size is about the same order as the thumbnails),
as more and more applications are providing high-res icons.

This patch is ported from mutter, cf
https://bugzilla.gnome.org/show_bug.cgi?id=689651

https://bugzilla.gnome.org/show_bug.cgi?id=504729
---
diff --git a/src/core/screen.c b/src/core/screen.c
index 52e020a..85d2165 100644
--- a/src/core/screen.c
+++ b/src/core/screen.c
@@ -119,14 +119,21 @@ set_wm_icon_size_hint (MetaScreen *screen)
 #define N_VALS 6
   gulong vals[N_VALS];
 
+  /* We've bumped the real icon size up to 96x96, but
+   * we really should not add these sorts of constraints
+   * on clients still using the legacy WM_HINTS interface.
+   */
+#define LEGACY_ICON_SIZE 32
+
   /* min width, min height, max w, max h, width inc, height inc */
-  vals[0] = META_ICON_WIDTH;
-  vals[1] = META_ICON_HEIGHT;
-  vals[2] = META_ICON_WIDTH;
-  vals[3] = META_ICON_HEIGHT;
+  vals[0] = LEGACY_ICON_SIZE;
+  vals[1] = LEGACY_ICON_SIZE;
+  vals[2] = LEGACY_ICON_SIZE;
+  vals[3] = LEGACY_ICON_SIZE;
   vals[4] = 0;
   vals[5] = 0;
-  
+#undef LEGACY_ICON_SIZE
+
   XChangeProperty (screen->display->xdisplay, screen->xroot,
                    screen->display->atom_WM_ICON_SIZE,
                    XA_CARDINAL,
diff --git a/src/include/common.h b/src/include/common.h
index 4ce800b..651fa07 100644
--- a/src/include/common.h
+++ b/src/include/common.h
@@ -258,8 +258,8 @@ struct _MetaButtonLayout
 };
 
 /* should investigate changing these to whatever most apps use */
-#define META_ICON_WIDTH 32
-#define META_ICON_HEIGHT 32
+#define META_ICON_WIDTH 96
+#define META_ICON_HEIGHT 96
 #define META_MINI_ICON_WIDTH 16
 #define META_MINI_ICON_HEIGHT 16
 
--
cgit v0.9.0.2
